tests/googleSsoTests.js

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright 2021-2022 Inclusive Design Research Centre, OCAD University
+ * All rights reserved.
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ */
+
+"use strict";
+
+require("json5/lib/register");
+const fluid = require("infusion");
+const axios = require("axios");
+const nock = require("nock");
+const url = require("url");
+const jqUnit = require("node-jqunit");
+
+require("../src/shared/driverUtils.js");
+require("./shared/utilsCommon.js");
+require("./shared/utilsSso.js");
+
+const config = require("./testConfig.json5");
+const serverUrl = "http://localhost:" + config.server.port;
+fluid.tests.utils.setDbEnvVars(config.db);
+
+const googleSso = require("../src/server/routes/ssoProviders/googleSso.js");
+const dbRequest = require("../src/server/ssoDbOps.js");
+
+jqUnit.module("Personal Data Server Google SSO Tests");
+
+fluid.registerNamespace("fluid.tests.googleSso");
+
+const mockAccessToken = {
+    access_token: "PatAccessToken.someRandomeString",
+    expires_in: 3600,
+    refresh_token: "anotherRandomString"
+};
+
+// Possible errors are "invalid_request", "invalid_client", "invalid_grant".
+// "unauthorized_client", "unsupported_grant_type", or "invalid_scope".
+// However, the status code is "400 Bad Request" for all of them -- use same
+// mock in all cases.
+// https://www.rfc-editor.org/rfc/rfc6749#section-5.2
+const mockErrorResponse = {
+    error: "invalid client",
+    error_description: "The specified client is unknown"
+};
+
+const mockUserInfo = {
+    id: "PatId",
+    name: "Pat Smith",
+    email: "pat.smith@somewhere.com",
+    locale: "en",
+    picture: "https://lh3.googleusercontent.com/picture/url",
+    given_name: "Pat",
+    family_name: "Smith",
+    verified_email: true
+};
+
+// Keep track of the payload returned by the auth request for consequent tests
+let authPayload;
+
+jqUnit.test("Google SSO tests", async function () {
+    jqUnit.expect(36);
+    try {
+        // Start the database
+        const dbStatus = await fluid.personalData.dockerStartDatabase(config.db.dbContainerName, config.db.dbDockerImage, config.db);
```

Comment:
It would be helpful to have a conditional here (perhaps controlled by an environment variable) which would skip this step and simply issue the test against a locally installed PostGres instance